src/psyclone/tests/domain/lfric/kernel/lfric_kernel_metadata_test.py

```diff
@@ -99,104 +112,79 @@ def test_init_args_error():
     '''
     with pytest.raises(ValueError) as info:
         _ = LFRicKernelMetadata(operates_on="invalid")
-    assert ("The operates_on metadata should be a recognised iteration "
-            "space (one of ['cell_column', 'domain', 'dof']) but found "
-            "'invalid'." in str(info.value))
+    print(str(info.value))
+    assert ("The OPERATES_ON metadata should be a recognised value "
+            "(one of ['cell_column', 'domain', 'dof']) but found 'invalid'."
+            in str(info.value))
 
-    with pytest.raises(ValueError) as info:
-        _ = LFRicKernelMetadata(procedure_name="1_invalid")
-    assert ("Expected procedure_name to be a valid Fortran name but found "
-            "'1_invalid'." in str(info.value))
+    with pytest.raises(TypeError) as info:
+        _ = LFRicKernelMetadata(shapes="invalid")
+    assert ("ShapesMetadata values should be provided as a list but found "
+            "'str'." in str(info.value))
 
-    with pytest.raises(ValueError) as info:
-        _ = LFRicKernelMetadata(name="1_invalid")
-    assert ("Expected name to be a valid Fortran name but found "
-            "'1_invalid'." in str(info.value))
+    with pytest.raises(TypeError) as info:
+        _ = LFRicKernelMetadata(evaluator_targets="invalid")
+    assert ("EvaluatorTargetsMetadata values should be provided as a list but "
+            "found 'str'." in str(info.value))
 
     with pytest.raises(TypeError) as info:
         _ = LFRicKernelMetadata(meta_args="error")
-    assert "meta_args should be a list but found str." in str(info.value)
+    assert ("MetaArgsMetadata values should be provided as a list but found "
+            "'str'." in str(info.value))
 
     with pytest.raises(TypeError) as info:
-        _ = LFRicKernelMetadata(meta_args=["error"])
-    assert ("meta_args should be a list of argument objects (of type "
-            "CommonArg), but found str." in str(info.value))
-    # TODO issue #1879 gh_shape, meta_funcs, meta_reference_element,
-    # meta_mesh
+        _ = LFRicKernelMetadata(meta_funcs="invalid")
+    assert ("MetaFuncsMetadata values should be provided as a list but found "
+            "'str'." in str(info.value))
 
+    with pytest.raises(TypeError) as info:
+        _ = LFRicKernelMetadata(meta_ref_element="invalid")
+    assert ("MetaRefElementMetadata values should be provided as a list but "
```

Comment:
It's possibly not the best that we have things named "Meta....Metadata" when something like "ArgsMetadata" or "FuncsMetadata" would be clearer. However, I think (if you agree) that should be a separate PR at this stage.

Reply:
It's because the metadata is called `meta_args`, `meta_funcs` etc., so I think it is right :-)

Reply:
It might be 'right' but I don't think it's nice :-) Still, it's a minor point.